### PR TITLE
Compress images before sending to Firebase Storage

### PR DESCRIPTION
**Pull request by lnogueir**
Prevents huge images from being uploaded to Firebase which caused some images to take multiple seconds to load on the frontend. Reduces image size by ~70%.

We can also compress videos with this library, which can be useful for stories.

Validated on android emulator and physical iPhone.

Ref:
https://www.npmjs.com/package/react-native-compressor

### CODE REVIEW

File: src/services/storageService.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 export async function uploadProfilePhoto(photoUri, userId, photoIdx) {
   return await uploadMediaAndGetUrl(photoUri, `${userId}/profile/${photoIdx}`);
 }
 
 async function uploadMediaAndGetUrl(uri, path) {
   const ref = storage().ref(path);
-  await ref.putFile(uri);
+  // TODO: once we add stories we can also compress videos with this library.
+  const compressedUri = await Image.compress(uri, {compressionMethod: 'auto'});
```
Comment (SultanEm) [inline src/services/storageService.js:11]:
```suggestion
  const compressedURI = await Image.compress(uri, {compressionMethod: 'auto'});
```
might be better

**Review by SultanEm — APPROVED** [review]

**Comment by jeverd** [comment]
approving, but have a question: do we have a loading indicator, when we upload to firebase?

**Review by jeverd — APPROVED** [review]

File: src/services/storageService.js (hunk 1 above)
Comment (lnogueir) [reply src/services/storageService.js:11]: On the function where this is defined (`uploadMediaAndGetUrl`) and in other places (like `GrpcService` or `GrpcProvider`) we're not uppercasing these acronyms, so I think it's best to just stick with not uppercasing it.

**Comment by lnogueir** [comment]
> approving, but have a question: do we have a loading indicator, when we upload to firebase?

I am giving the responsibility to add loading indicator logic to the user of this function. Rn it's only being used on the onboarding photos screen. There we have a loading indicator while this function is executing.

**Pull request merged**